### DISCUSSION TITLE
How to use ODBC?

### DISCUSSION THREAD

**Discussion by Tyrben** [pocoproject/poco#3532] [Q&A] [answered]
In this part of the lib `Poco::Data::ODBC`, I'm not figuring out if I am missing something or if some of the documented functions are simply not implemented in the lib.

I'm speaking of that : https://docs.pocoproject.org/current/Poco.Data.ODBC.html
Especially (but not only) the function `SQLTables(...)`.
When I use it, it looks to be missing, I don't get any response, but a nice exception.
Doc: https://docs.pocoproject.org/current/Poco.Data.ODBC.html#3649

Here is the ref to the original function with the same signature in the Windows ODBC API
https://docs.microsoft.com/en-us/sql/odbc/reference/syntax/sqltables-function?view=sql-server-ver15

-   How do you use ODBC in Poco in general? Do I need to change my approach?
-   Do I need to reimplement something, or inherit and specialize?
-   Do you have any example using this functions?

An example of code, (just set a valid Connexion String, of course)

```cpp
#include <iostream>
#include <Poco/Data/SessionFactory.h>
#include <Poco/Data/ODBC/Connector.h>

using namespace Poco::Data;

void main() {
	try {
		ODBC::Connector::registerConnector();
		Session session{ SessionFactory::instance().create(ODBC::Connector::KEY, "Driver={IBM i Access ODBC Driver};SYSTEM=192.168.0.10;DBQ=BASENAME;UID=user;PWD=passwd") };
		if (session.isConnected()) {
			StatementImpl *hstmt = session.createStatementImpl();
			ODBC::SQLTables(hstmt, NULL, 0, NULL, 0, NULL, 0, NULL, 0);
		}
	}
	catch (const Poco::Exception& err) {
		std::cerr << "Error thrown: " << err.displayText() << "\n";
	}
}
```

Resulting to the message:
> "Error thrown: Not implemented"

**Comment by aleks-f** [comment, marked as answer]
See [POCO Data User Guide](https://docs.pocoproject.org/current/00200-DataUserManual.html)
